src/router.js

```diff
@@ -157,12 +162,8 @@ export function Router(props) {
 			return true;
 		}
 		return false;
-	}, [url]);
+	}, [url, JSON.stringify(matchProps)]);
```

Comment:
@JoviDeCroock Have any opinions on what we do here? Obviously `JSON.stringify` ain't ideal though I'm not sure if we want to start doing deep comparisons ourselves or with a dep either.

As the updated test cases show, your changes in #12 have us rendering routes more than we should which sets up a dependency on the user-passed props.